src/test/java/io/confluent/connect/jdbc/sink/integration/PostgresDatatypeIT.java

```diff
@@ -42,67 +47,167 @@
+    props.put(DLQ_TOPIC_REPLICATION_FACTOR_CONFIG, "1");
+    props.put(MAX_RETRIES, "2");
+
+    createTableWithPrimaryKey();
+    connect.configureConnector("jdbc-sink-connector", props);
+    waitForConnectorToStart("jdbc-sink-connector", 1);
+
+    final Schema schema = SchemaBuilder.struct().name("com.example.Person")
+        .field("firstname", Schema.STRING_SCHEMA)
+        .field("lastname", Schema.STRING_SCHEMA)
+        .build();
+    final Struct firstStruct = new Struct(schema)
+        .put("firstname", "Christina")
+        .put("lastname", "Brams");
+
+    produceRecord(schema, firstStruct);
+    // Send the same record for a PK collision
+    produceRecord(schema, firstStruct);
+
+    // Now, create and send another normal record
+    Struct secondStruct = new Struct(schema)
+        .put("firstname", "Brams")
+        .put("lastname", "Christina");
+
+    produceRecord(schema, secondStruct);
+
+    waitForCommittedRecords("jdbc-sink-connector", Collections.singleton(tableName), 3, 1,
+        TimeUnit.MINUTES.toMillis(3));
+
+    KafkaConsumer<byte[], byte[]> consumer =
+        connect.kafka().createConsumerAndSubscribeTo(Collections.emptyMap(), DLQ_TOPIC_NAME);
+
+    ConsumerRecords<byte[], byte[]> records =
+        consumer.poll(Duration.ofMillis(CONSUME_MAX_DURATION_MS));
+
+    assertEquals(1, records.count());
   }
 
   @Test
-  public void testWriteToTableWithUuidColumn() throws SQLException {
+  public void testRecordSchemaMoreFieldsThanTableSendsToErrorReporter() throws Exception {
+    props.put(ERRORS_TOLERANCE_CONFIG, ToleranceType.ALL.value());
+    props.put(DLQ_TOPIC_NAME_CONFIG, DLQ_TOPIC_NAME);
+    props.put(DLQ_TOPIC_REPLICATION_FACTOR_CONFIG, "1");
+
+    createTableWithLessFields();
+    connect.configureConnector("jdbc-sink-connector", props);
+    waitForConnectorToStart("jdbc-sink-connector", 1);
+
+    final Schema schema = SchemaBuilder.struct().name("com.example.Person")
+        .field("firstname", Schema.STRING_SCHEMA)
+        .field("lastname", Schema.STRING_SCHEMA)
+        .field("jsonid", Schema.STRING_SCHEMA)
+        .field("userid", Schema.STRING_SCHEMA)
+        .build();
+    final Struct struct = new Struct(schema)
+        .put("firstname", "Christina")
+        .put("lastname", "Brams")
+        .put("jsonid", "5")
+        .put("userid", UUID.randomUUID().toString());
+
+    String kafkaValue = new String(jsonConverter.fromConnectData(tableName, schema, struct));
+    connect.kafka().produce(tableName, null, kafkaValue);
+
+    waitForCommittedRecords("jdbc-sink-connector", Collections.singleton(tableName), 1, 1,
+        TimeUnit.MINUTES.toMillis(2));
+
+    KafkaConsumer<byte[], byte[]> consumer =
+        connect.kafka().createConsumerAndSubscribeTo(Collections.emptyMap(), DLQ_TOPIC_NAME);
+
+    ConsumerRecords<byte[], byte[]> records =
+        consumer.poll(Duration.ofMillis(CONSUME_MAX_DURATION_MS));
+
+    assertEquals(1, records.count());
```

Comment:
Same question here w/r/t manually constructing the consumer.

Reply:
Fixed this.